### PR TITLE
Predis 1.0 in branch 1.1

### PR DESCRIPTION
**Pull request by theUniC**
Hi,

I've reproduced all the work to integrate Predis 1.0 into the ```1.1``` branch.

### CODE REVIEW

**Comment by bastnic** [comment]
:+1:

**Comment by michaelthieulin** [comment]
+1

**Comment by michaelthieulin** [comment]
:+1:

**Comment by theUniC** [comment]
Do I miss something left to do?

**Comment by ajanssen** [comment]
+100

**Comment by ternel** [comment]
Nice :+1:

File: composer.json [hunk 1]
```diff
@@ -30,5 +31,8 @@
     "autoload": {
```
Comment (damienalexandre) [inline composer.json]: You also needs to change the suggest:

     "predis/predis": "If you want to use predis (currently only v0.8.x is supported).",

File: Tests/DependencyInjection/SncRedisExtensionTest.php [hunk 2]
```diff
@@ -198,7 +198,7 @@ public function testClientProfileOption()
         $options = $container->getDefinition('snc_redis.client.default_options')->getArgument(0);
 
         $this->assertSame((float) 2, $config['clients']['default']['options']['profile'], 'Profile version 2.0 was parsed as float');
-        $this->assertSame('Predis\\Profile\\ServerVersion20', $profileDefinition->getClass(), 'Profile definition is instance of Predis\\Profile\\ServerVersion20');
+        $this->assertSame('Predis\\Profile\\RedisVersion200', $profileDefinition->getClass(), 'Profile definition is instance of Predis\\Profile\\ServerVersion20');
```
Comment (damienalexandre) [inline Tests/DependencyInjection/SncRedisExtensionTest.php]: Test description is not updated.

**Comment by theUniC** [comment]
Sorry for the delay. I've already fixed composer.json and the test description.

**Comment by TomAdam** [comment]
@snc would be great to get this merged.

**Comment by badlamer** [comment]
+1

**Comment by fdeweger** [comment]
+1

**Comment by piotrantosik** [comment]
Any update on this?

**Comment by mmoreram** [comment]
:+1: Any update on that?

**Comment by mmoreram** [comment]
@snc Would be really nice, if you are no longer available for the maintenance of this project, to create a new organization, adding both project `srcRedis` and `sncRedisBundle` and allowing other people to do that essential work.

Please man, do that...

**Comment by dominics** [comment]
@mmoreram bravo

@snc or at least remove the 'project: maintained' badge so people know what they're getting themselves into. There are serious isses with open PRs over six months old (cross-connection writes, etc.).

**Comment by Seldaek** [comment]
I sent an email to snc because he might not get github notifications in his inbox.

**Comment by snc** [comment]
Oh damn, big sorry! Thanks for mailing me @Seldaek! I'll merge this now.

**Comment by mmoreram** [comment]
Thanks @Seldaek

**Comment by mmoreram** [comment]
@snc would be really nice to tag `1.1.9` in order to create stable dependencies between projects :)

**Comment by snc** [comment]
@mmoreram Just pushed the tag.

**Comment by mmoreram** [comment]
@snc Thanks :)

**Comment by javihgil** [comment]
Hi, I've just update my composer project (configured package is 1.1.*) and I get this error after updating to 1.1.9:

PHP Fatal error:  Class 'Predis\Profile\Factory' not found in vendor/snc/redis-bundle/Snc/RedisBundle/DependencyInjection/SncRedisExtension.php on line 158

**Comment by Seldaek** [comment]
You need to require predis `^1.0` to use 1.1.9.. which is kind of a BC break. I don't think this should have been released as is.. If the code worked with predis 1.0 or 0.8 it would be alright I guess, but that doesn't seem to be possible.

**Comment by javihgil** [comment]
Thanks, that solves my problem

**Comment by digitalkaoz** [comment]
im on @Seldaek's side..ugly avoidable BC break which could be simple solved by a `class_exists`

**Comment by snc** [comment]
@digitalkaoz Are you sure a simple `class_exists` would solve this? I'm very busy right now and don't have the time to check this... I'm thinking about reverting this PR and instead merge it into the `master` branch.

**Comment by Seldaek** [comment]
class_exists isn't good enough IMO, as you are extending classes that 
got renamed.. But yes doing this in a patch version isn't very nice :) 
Doing it in master would be a start but the question is can we safely 
upgrade to master or is it breaking everything?

**Comment by digitalkaoz** [comment]
i dont know i guess this might work?

```php
if (class_exists(\Predis\Profile\ServerProfile::class)) {
    $profileDef = new Definition(get_class(\Predis\Profile\ServerProfile::get($client['options']['profile'])));
else {
   $profileDef = new Definition(get_class(\Predis\Profile\Factory::get($client['options']['profile'])));
}
```

wouldnt that work? or even vice versa, some the both class didnt exists the past right?

**Comment by abraovic** [comment]
Hi @snc, 

you might update doc https://github.com/snc/SncRedisBundle/blob/master/Resources/doc/index.md for predis version. It is still 0.8.x-dev which causes this problems.

Best,
Ante.

**Comment by liuggio** [comment]
In production i had to patch this bundle using the @digitalkaoz solution
Easy to reproduce the website is https://poser.pugx.org 
agaist this repo  https://github.com/liuggio/badge-poser/tree/flat-square
/cc @snc

**Pull request merged**